crates/core/main.rs

```diff
@@ -68,11 +68,22 @@ fn try_main(args: Args) -> Result<()> {
         process::exit(0)
     } else if messages::errored() {
         process::exit(2)
+    } else if !matched {
+        eprint_nothing_searched();
+        process::exit(1)
```

Comment:
Hmmm, this is definitely not correct. :-( This is printing a message when nothing is _matched_, but what we want to do is print a message when nothing is _searched_.

Reply:
Oh, I see. I can't seem to find the logic where the error message used to be shown, i.e. where we can conclude that there are no search candidates. Do you happen to know the area so I can hone in on it? I'm mostly looking in `walk.rs` right now.

Reply:
I would keep trying in `main.rs`. In particular, I believe the logic will need to be in both the `search` and `search_parallel` routines. For the former, I think it needs to check if `searcher.search()` is ever executed. Similarly for the latter.